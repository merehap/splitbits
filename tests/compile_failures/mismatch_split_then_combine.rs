use splitbits::*;

fn main() {
    let _ = splitbits_then_combine!(0b1001_1010, "aaab bbbb", "bbba aaaa");
}
