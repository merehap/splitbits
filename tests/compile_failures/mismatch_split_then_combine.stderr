error: proc macro panicked
 --> tests/compile_failures/mismatch_split_then_combine.rs:4:13
  |
4 |     let _ = splitbits_then_combine!(0b1001_1010, "aaab bbbb", "bbba aaaa");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: assertion `left == right` failed: The width of field 'b' must match between the input templates and the output template.
            left: 5
           right: 3
